pkg/webhook/validation/podspec_dryrun.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2020 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/logging"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	"knative.dev/serving/pkg/reconciler/revision/resources"
+)
+
+func decodeTemplateAndValidate(ctx context.Context, val interface{}, namespace string) error {
+	templ := &v1.RevisionTemplateSpec{}
+	asData := val.(map[string]interface{})
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(asData, templ); err != nil {
```

Comment:
If converter is like the `encoding/json` a weird input can leave you an empty `RevisionTemplateSpec`  but not report an error ie. `{}` or `null` maybe

I'm not clear how webhooks are ordered so there may not be a guarantee that this was validated prior.  We should probably call `Validate` to make sure the template is valid?

Reply:
Added a nil or empty check